### PR TITLE
Adds Humanity, a stage 4 virus syndrome.

### PR DESCRIPTION
**Pull request by Irradiation**
It turns non-humans into humans.

[content]

:cl:
 * rscadd: Adds Humanity, a stage 4 virus syndrome. It turns non-humans into Humans.

### CODE REVIEW

**Comment by MadmanMartian** [comment]
Why are you checking mob type and not just ishuman && !isjusthuman ?

Then you would get everything that is /mob/living/carbon/human(oid), but they aren't actually humans, so now your virus actually works.

image

**Comment by Irradiation** [comment]
The virus actually worked prior to your comment, though. I'm making some adjustments to work on every human(oid) specie.

**Comment by boy2mantwicethefam** [comment]
cat

**Comment by MadmanMartian** [comment]
It worked because you were spawning in those subtypes. A normal grey would not have been affected as they are spawned in as a /mob/living/carbon/human and then their species is changed to grey and their icons updated.

**Comment by Irradiation** [comment]
That sounds very gay, then. Time to make changes.

**Comment by Irradiation** [comment]
I'm going to try out your solution, but in the meantime, the new one I committed seems to be working on late-join non-humans and those modified by VV or Voxpox.

**Comment by MadmanMartian** [comment]
Your current version does work, yes. Sorry, I was still talking about your first itteration.

Rather than checking the name of the species, check the type of the species (!istype(species, /datum/species/human). Easier to maintain that way.

**Comment by Gazathor** [comment]
ability to extract their newly found humanity whence

**Comment by 9percents** [comment]
Virology: Become Human

**Comment by Irradiation** [comment]
Implemented R0b0's suggestion and cut down a redundant line.

File: code/modules/virus2/effect/stage_4.dm [hunk 1]
```diff
@@ -913,13 +913,12 @@
 	name = "Forced Humanity Syndrome"
 	desc = "A recent development by human supremacists. Causes non-human infected to mutate into a Human."
 	stage = 4
-	badness = EFFECT_DANGER_DEADLY
+	badness = EFFECT_DANGER_HELPFUL
```
Comment (MadmanMartian) [inline code/modules/virus2/effect/stage_4.dm]: 🤔

Comment (Irradiation) [reply code/modules/virus2/effect/stage_4.dm]: Darn it, he looked.

Seriously though, it doesn't have any business being at deadly.

**Comment by bathosbathtime** [comment]
There is a vague danger of someone gaining humanity in an EVA situation wearing a suit that can not be worn by humans. While I'm not against it being considered 'helpful' I do want to point out there are some dangers in changing someone's species to human in game. Though switching off internals that are toxic or at least non-breathable to humans is more of an end user issue.

**Comment by MadmanMartian** [comment]
What if it m2h'd monkeys?

**Pull request merged**